core/arch/arm/kernel/entry_a32.S

```diff
@@ -551,10 +553,19 @@ shadow_stack_access_ok:
 	ldr	r0, =threads
 	ldr	r0, [r0, #THREAD_CTX_STACK_VA_END]
 	mov	sp, r0
+	bl	__get_core_pos
```

Comment:
Would it make sense to call `thread_get_core_local()` instead?

Reply:
Totally. Fixed in new version.